### ISSUE TITLE
`--field-separator` does not work correctly if no newline at EOF

### ISSUE THREAD

**Issue by taotao54321** [theryangeary/choose#48]
version: commit 97b3abf5
environment: linux x86_64

When an input has no newline at EOF, `--field-separator` option does not work correctly.

```sh
$ /bin/echo 'a:b' | target/debug/choose -f ':' 1
b

$ /bin/echo -n 'a:b' | target/debug/choose -f ':' 1
(expected "b", but no output)
```

Moreover, if an input contains non-ASCII characters, `choose` can panic:

```sh
$ /bin/echo -n 'あ:い' | target/debug/choose -f ':' 1
thread 'main' panicked at 'byte index 6 is not a char boundary; it is inside 'い' (bytes 4..7) of `あ:い`', src/main.rs:77:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I guess this code causes the issue:
https://github.com/theryangeary/choose/blob/97b3abf5204badf60eb79be6be07285bdef83049/src/main.rs#L76-L80

This code seems to trim the last character unconditionally, but it will have to check if a line actually has a newline.

**Comment by theryangeary** [comment]
Hi, thanks for reporting. I will investigate this further and let you know when a fix is available.

**Comment by theryangeary** [comment]
The issue has been resolved by #50, and the fix is available in `develop`.

**Comment by taotao54321** [comment]
I confirmed v1.3.4 works well with my usecase, including non-ASCII characters.

Thank you!
